dask_ml/model_selection/_hyperband.py

```diff
@@ -24,59 +26,73 @@ def _get_nr(R, eta=3):
     return list(map(int, N)), list(map(int, R)), brackets
 
 
-def _partial_fit(model_and_meta, X, y, meta=None, fit_params={},
+def _partial_fit(model_meta_future, X, y, meta=None, fit_params={},
                  dry_run=False, **kwargs):
+    start = time()
     assert isinstance(X, np.ndarray)
     assert y is None or isinstance(y, np.ndarray)
-    model, m = model_and_meta
+    model, m = model_meta_future
     if meta is None:
         meta = m
     while meta['iterations'] < meta['partial_fit_calls']:
         if not dry_run:
             model.partial_fit(X, y, **fit_params)
         meta["iterations"] += 1
+    meta['mean_fit_time'] += time() - start
     return model, meta
 
 
 def _score(model_and_meta, x, y, dry_run=False):
+    start = time()
     model, meta = model_and_meta
+    #  seed = hash(((k, v) for k, v in meta.items())) % 2**32
+    #  rng = np.random.RandomState(seed)
     if dry_run:
+        #  score = rng.rand()
         score = 0
     else:
         # TODO: change this to dask_ml.get_scorer
         score = model.score(x, y)
     meta.update(score=score)
+    assert meta['iterations'] > 0
+    meta['mean_score_time'] += time() - start
+    meta['mean_test_score'] = score
     return meta
 
 
 def _top_k(results, k=1, sort="score"):
-    return sorted(results, key=lambda x: x[sort])[-k:]
+    res = sorted(results, key=lambda x: x[sort])[-k:]
+    assert len(res) == k
+    return res
 
 
-def _to_promote(result, completed_jobs, eta=None):
+def _to_promote(result, completed_jobs, eta=None, asynchronous=None):
```

Comment:
So far this keyword seems like it doesn't do anything.  Should we keep it?

Reply:
There is an asynchronous extension to Hyperband. I'm planning on putting it in this PR because it's simple to implement (in fact, simpler than the synchronous version).

Reply:
Do we want both?  What is your time estimate on the other version?  It would be good to close something out soonish.